packages/core/test/request-builder/header-builder/auth-headers.spec.ts

```diff
@@ -0,0 +1,79 @@
+/* Copyright (c) 2020 SAP SE or an SAP affiliate company. All rights reserved. */
+import { MapType } from '@sap-cloud-sdk/util';
+import { buildHeadersForDestination, Destination } from '../../../src';
+import { muteLoggers } from '../../test-util/mute-logger';
+import { buildAuthorizationHeader } from '../../../src/request-builder/header-builder/auth-headers';
+
+describe('Authorization header builder', () => {
+  beforeAll(() => {
+    muteLoggers('auth-headers');
+  });
+
+  it('does not throw on NoAuthentication', async () => {
+    await expect(
+      buildAuthorizationHeader({
+        url: 'https://example.com',
+        authentication: 'NoAuthentication'
+      })
+    ).resolves.not.toThrow();
+  });
+
+  it('does not throw on ClientCertificateAuthentication', async () => {
+    await expect(
```

Comment:
Is there an other test checking the content expected as the header field if auth type is `ClientCertificateAuthentication`? If not I would add an assertion and not only chcek that it does not throw.

Reply:
I agree on this, but for all tests. I only adjusted the tests (and added some for code that I added), but I didn't want to also refactor our tests. This implicitly tests that headers are set and therefore built correctly, so it is testing the correct thing at least. As mentioned before, I created a ticket for formalizing and improving our test strategy.

Reply:
Created a ticket for this